gcloud/pubsub/subscription.py

```diff
@@ -67,12 +75,16 @@ def from_api_repr(cls, resource, client, topics=None):
         if topics is None:
             topics = {}
         topic_path = resource['topic']
-        topic = topics.get(topic_path)
-        if topic is None:
-            # NOTE: This duplicates behavior from Topic.from_api_repr to avoid
-            #       an import cycle.
-            topic_name = topic_name_from_path(topic_path, client.project)
-            topic = topics[topic_path] = client.topic(topic_name)
+        if topic_path == cls._DELETED_TOPIC_PATH:
+            # Use a name which cannot conflict ('#' not allowed).
+            topic = client.topic(None)
```

Comment:
> Because of the delegation. Not for other reasons?

A subscription is fundamentally related to a topic:  it isn't just incidental.  E.g., when calling the `create` API, the `topic` key (which we derive from `self.topic.path`) is not optional.

> Worth having an explicit subclass representing a deleted topic?

I wouldn't invent a subclass `Topic` just for this edge case.

> Or just another container for the client?

We would need to copy out the `project` and `client` from the `topic` instance passed to `Subscription.__init__`, and then add extra arguments to allow passing those values in in the edge case that there was no actual topic.  I think we're better off just creating a brain-damaged `Topic` instance.